internal/pkg/mqtt/client_test.go

```diff
@@ -217,6 +255,206 @@ func TestInvalidClientOptions(t *testing.T) {
+				mqtt.KeyPEMBlock:  "KeyPEMBlock",
+			},
+			certCreator:     mockCertCreator(nil),
+			certLoader:      mockCertLoader(nil),
+			expectError:     false,
+			expectTLSConfig: true,
+		},
+		{
+			name: "Skip TLS Config from PEM Block for non-supported TLS protocols",
+			hostConfig: types.HostInfo{
+				Host:     "localhost",
+				Protocol: "tcp",
+			},
+			optionalConfig: map[string]string{
+				mqtt.CertPEMBlock: "CertPEMBlock",
+				mqtt.KeyPEMBlock:  "KeyPEMBlock",
+			},
+			certCreator:     mockCertCreator(nil),
+			certLoader:      mockCertLoader(nil),
+			expectError:     false,
+			expectTLSConfig: false,
+		},
+		{
+			name:       "Fail Create TLS Config from PEM File",
+			hostConfig: TlsHostInfo,
+			optionalConfig: map[string]string{
+				mqtt.CertPEMBlock: "CertPEMBlock",
+				mqtt.KeyPEMBlock:  "KeyPEMBlock",
+			},
+			certCreator:     mockCertCreator(errors.New("test error")),
+			certLoader:      mockCertLoader(nil),
+			expectError:     true,
+			expectTLSConfig: false,
+		},
+		{
+			name:       "Load TLS Config from Cert File",
+			hostConfig: TlsHostInfo,
+			optionalConfig: map[string]string{
+				mqtt.CertFile: "./cert",
+				mqtt.KeyFile:  "./key",
+			},
+			certCreator:     mockCertCreator(nil),
+			certLoader:      mockCertLoader(nil),
+			expectError:     false,
+			expectTLSConfig: true,
+		},
+		{
+			name:       "Skip Load TLS Config from Cert File for un-supported protocols",
+			hostConfig: types.HostInfo{Host: "localhost", Protocol: "tcp"},
+			optionalConfig: map[string]string{
+				mqtt.CertFile: "./cert",
+				mqtt.KeyFile:  "./key",
+			},
+			certCreator:     mockCertCreator(nil),
+			certLoader:      mockCertLoader(nil),
+			expectError:     false,
+			expectTLSConfig: false,
+		},
+		{
+			name:       "Fail Load TLS Config from Cert File",
+			hostConfig: TlsHostInfo,
+			optionalConfig: map[string]string{
+				mqtt.CertFile: "./cert",
+				mqtt.KeyFile:  "./key",
+			},
+			certCreator:     mockCertCreator(nil),
+			certLoader:      mockCertLoader(errors.New("test error")),
+			expectError:     true,
+			expectTLSConfig: false,
+		},
+		{
+			name:       "Fail Load TLS Config For Invalid Options",
+			hostConfig: TlsHostInfo,
+			optionalConfig: map[string]string{
+				mqtt.Qos: "abc",
+			},
+			certCreator:     mockCertCreator(nil),
+			certLoader:      mockCertLoader(errors.New("test error")),
+			expectError:     true,
+			expectTLSConfig: false,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			client, err := NewMQTTClientWithCreator(
+				types.MessageBusConfig{
+					PublishHost: test.hostConfig,
+					Optional:    test.optionalConfig,
+				},
+				json.Marshal,
+				json.Unmarshal,
+				ClientCreatorWithCertLoader(test.certCreator, test.certLoader))
+
+			if test.expectError && err == nil {
+				t.Error("Expected error but did not observe one")
+				return
+			}
+
+			if test.expectError && err != nil {
+				return
+			}
+
+			if err != nil {
```

Comment:
Is this the correct logic? This it should be `==`? but check not needed if use my suggestion above.